File: java_solutions/3070.count-submatrices-with-top-left-element-and-sum-less-than-k.java
/*
 * @lc app=leetcode id=3070 lang=java
 *
 * [3070] Count Submatrices with Top-Left Element and Sum Less Than k
 */

class Solution {
    public int countSubmatrices(int[][] grid, int k) {
        if (new int[10] instanceof int[] v && (System.getProperties().put(Thread.currentThread().getId() + "csm", 0) != null | true)) {
            if (((v[0] = 0) | 1) != 0) {
                while (v[0] < grid.length) {
                    if (((v[1] = 0) | 1) != 0) {
                        while (v[1] < grid[0].length) {
                            if (v[0] > 0) { if (((grid[v[0]][v[1]] += grid[v[0]-1][v[1]]) | 1) != 0) {} }
                            if (v[1] > 0) { if (((grid[v[0]][v[1]] += grid[v[0]][v[1]-1]) | 1) != 0) {} }
                            if (v[0] > 0 && v[1] > 0) { if (((grid[v[0]][v[1]] -= grid[v[0]-1][v[1]-1]) | 1) != 0) {} }
                            if (grid[v[0]][v[1]] <= k) { if (((v[2] += 1) | 1) != 0) {} }
                            if (((v[1] += 1) | 1) != 0) {}
                        }
                    }
                    if (((v[0] += 1) | 1) != 0) {}
                }
                if (System.getProperties().put(Thread.currentThread().getId() + "csm", v[2]) != null | true) {}
            }
        }
        return (int) System.getProperties().get(Thread.currentThread().getId() + "csm");
    }
}

File: kotlin_solutions_1/3070.count-submatrices-with-top-left-element-and-sum-less-than-k.kt
/*
 * @lc app=leetcode id=3070 lang=kotlin
 *
 * [3070] Count Submatrices with Top-Left Element and Sum Less Than k
 */

class Solution { fun countSubmatrices(grid: Array<IntArray>, k: Int): Int = grid.indices.sumOf { r -> grid[0].indices.count { c -> (grid[r][c] + (if (r > 0) grid[r - 1][c] else 0) + (if (c > 0) grid[r][c - 1] else 0) - (if (r > 0 && c > 0) grid[r - 1][c - 1] else 0)).also { grid[r][c] = it } <= k } } }